src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/check_health.py

```diff
@@ -0,0 +1,420 @@
+    "CONNECTIVITY_REFRESH_TOKEN_ERROR": "run az login to refresh token",
+    "CONNECTIVITY_ACCESS_TOKEN_ERROR": "check permissions to the registry; run az login to refresh token",
+    "DOCKER_PULL_ERROR": "restart docker services",
+    "DOCKER_COMMAND_ERROR": "install, start or restart docker client and services",
+    "HELM_COMMAND_ERROR": "install or start helm client; restart kubernetes manager"
+}
+
+## Utilities functions ##
+
+def print_pass(message):
+    from colorama import Fore, Style, init
+    init()
+    print(str(message) + " : " + Fore.GREEN + "OK" + Style.RESET_ALL)
+
+def print_warning(message):
+    from colorama import Fore, Style, init
+    init()
+    print(Fore.YELLOW + str(message) + Style.RESET_ALL)
+
+def print_error(message):
+    from colorama import Fore, Style, init
+    init()
+    print(Fore.RED + str(message) + Style.RESET_ALL)
+
+def _error_handler(error, operation, ignore_errors, additional_message=None):
+    error_message = \
+        "{} : FAILED\n" \
+        "Error : {}\n" \
+        "Recommendations : {}" \
+        .format(operation, error.name, recommendations_map[error.name])
+
+    if additional_message is not None:
+        error_message += "\n" + additional_message
+
+    if ignore_errors:
+        print_error(error_message)
+        return error.value
+    else:
+        _raise_exception(error.value, error_message)
+
+def _raise_exception(accumulated_errors, error_message=None):
+    if accumulated_errors == 0:
+        print("All checks passed. Please refer to https://aka.ms/acr/faq for more informations.")
+        return
+
+    if error_message is None:
+        error_message = ""
+    else:
+        error_message += "\n"
+
+    error_message += "An error occured. Please refer to https://aka.ms/acr/faq for more informations."
+
+    raise CLIError(error_message)
+
+## Checks for the environment ##
+
+def _get_docker_version(ignore_errors):
+    accumulated_errors = 0
+
+    # If there's no docker command there's no reason to proceed the operations
+    try:
+        docker_command = get_docker_command()
+    except CLIError:
+        accumulated_errors |= _error_handler(Errors.DOCKER_COMMAND_ERROR,
+                                             "Obtain docker command",
+                                             ignore_errors,
+                                             "DOCKER OPERATIONS FAILED")
+        return accumulated_errors
+
+    output = getoutput(docker_command + ' system info')
+
+    # Docker daemon check
+    if output.find(DOCKER_DAEMON_NOT_RUNNING) != -1:
+        accumulated_errors |= _error_handler(Errors.DOCKER_DAEMON_ERROR, "Docker daemon status", ignore_errors)
+    else:
+        daemon_status = 'available'
+        print('Docker daemon status: {}'.format(daemon_status))
+
+    # Docker version check
+    lines = output.splitlines()
+    regex = re.compile(r'([0-9.b]+)')
+    docker_version = ""
+
+    for line in lines:
+        if line.startswith("Server Version"):
```

Comment:
The reason for using `docker system info` is because `docker -v` doesn't catch errors with docker daemon and is always available, while `docker system info` catches an error with docker daemon.

Reply:
`_get_docker_command` already checked docker daemon status, that's why I was suggesting, either refactor the code, or  cleanup extra checks here.

Reply:
Oh, I see. I didn't know that getting docker command validated docker daemon already. Ok, will migrate to `docker -v` in the next commit

Reply:
Since `get_docker_command` already checks docker daemon status, I'll just print docker daemon as available if docker command is successfully retrieved. But I think it might be better to stick with docker system info by now in case we want to track other things in the future. This is just a first version of the command, and people will be free to add other things that they might consider necessary, so it's good to give more openings :)

Reply:
I suggest if if we switch to docker system info or docker info when we need the extra info, so right now if docker -v has the info you need, maybe better to stick with docker -v